blaze-client/src/main/scala/org/http4s/blaze/client/Connection.scala

```diff
@@ -33,4 +35,7 @@ private[client] trait Connection[F[_]] {
 
   /** The key for requests we are able to serve */
   def requestKey: RequestKey
+
+  /** Deadline for the connection to be borrowed before we evict it */
+  @volatile var borrowDeadline: Option[Deadline] = None
```

Comment:
I'm wondering if the `borrowDeadline` is a property of the `Connection` actually. Or maybe the `PoolManager` should own the fact when a connection needs to borrow. 🤔 WDYT?

Reply:
I think it needs to be _associated_ with a `Connection`, but it could be the `PoolManager` that maintains the association.  It would need to wrap some metadata around the connections to do so.  It could probably get rid of the mutation then. 🤔